stdlib/public/core/Collection.swift

```diff
@@ -1671,3 +1671,70 @@ extension Collection where SubSequence == Self {
     self = self[idx..<endIndex]
   }
 }
+
+extension Collection {
+
+  @inlinable // protocol-only
+  @warn_unqualified_access
+  public func minIndex(
+    by areInIncreasingOrder: (Element, Element) throws -> Bool
+  ) rethrows -> Index? {
+    guard !isEmpty else { return nil }
+    var resultIndex = startIndex
+    for index in indices.dropFirst() {
+      if try areInIncreasingOrder(self[index], self[resultIndex]) { resultIndex = index }
+    }
+    return resultIndex
+  }
+
+  @inlinable // protocol-only
+  @warn_unqualified_access
+  public func maxIndex(
+    by areInIncreasingOrder: (Element, Element) throws -> Bool
+  ) rethrows -> Index? {
+    guard !isEmpty else { return nil }
+    var resultIndex = startIndex
+    for index in indices.dropFirst() {
+      if try areInIncreasingOrder(self[resultIndex], self[index]) { resultIndex = index }
+    }
+    return resultIndex
+  }
+
+  @inlinable // protocol-only
+  @warn_unqualified_access
+  public func minmaxIndices(
+    by areInIncreasingOrder: (Element, Element) throws -> Bool
+  ) rethrows -> (Index, Index)? {
```

Comment:
Would it be better if the elements of the result tuple were labelled? Please discuss this point in the proposal.